src/lib/Libwin/passwd.c

```diff
@@ -5036,12 +5016,15 @@ cache_usertoken_and_homedir(char *user,
 	if (user == NULL)
 		return;
 
-	if(user_num == 0)
+	if(passwd_cache_init == 0){
```

Comment:
Why do the CLEAR_HEAD here at all? Why not just leave it to pbsd_main/init, where it gets initialized anyway?

Reply:
It gets initialised there after server startup, but these functions are called from pbs_account and pbs_ds_monitor as well. While testing these changed I saw both these crashing if I wasn't initialising the list here.

Reply:
Ah ok, i see. Thanks for explaining that. In that case, it makes sense the way you had done it earlier. Since you are anyway going to have the "check and initialize" in the individual functions itself, there is no need to also add it to pbsd_main.c. Let's do only one kind of initialization. Otherwise it looks good.

Reply:
Okay, will revert those changes.